src/vsr/replica.zig

```diff
@@ -363,9 +363,23 @@ pub fn ReplicaType(
             for (self.journal.headers) |*header| {
                 if (header.command == .prepare and header.op > op_head) {
                     assert(self.log_view >= header.view);
-                    assert(self.log_view == self.view);
-
-                    op_head = header.op;
+                    // Typically if there is an op in the WAL higher than the durable headers'
+                    // head op, we must have crashed with a durable SV, not a durable DVC.
+                    //
+                    // There is one exception:
+                    // 1. New-primary sends (and persists) a DVC: 5,6,7.
+                    // 2. New-primary receives DVC quorum. Suffix is 6,7,8.
+                    //    It already has 6,7. op=8 is new to us, but part of our prior log_view.
+                    // 3. New-primary repairs 8 (writes it to the WAL).
+                    // 4. New-primary crashes/recovers — op=8 was part of the same log_view, so the
+                    //    journal doesn't truncate it, and the primary recovers with a new head.
+                    // To avoid special-casing this all over, we pretend this higher op doesn't
+                    // exist. This is safe because the prior view-change didn't complete.
+                    assert(self.log_view <= self.view);
+                    assert(self.log_view == self.view or
+                        self.replica == self.primary_index(self.view));
+
+                    if (self.log_view == self.view) op_head = header.op;
```

Comment:
Briefly thought if 

```zig
                    if (self.log_view == self.view) {
                        op_head = header.op;
                    } else {
                        assert(self.replica == self.primary_index(self.view));
                        // Typically if there is an op in the WAL higher than the durable headers'
                        // head op, we must have crashed with a durable SV, not a durable DVC.
                        //
                        // There is one exception:
                        // 1. New-primary sends (and persists) a DVC: 5,6,7.
                        // 2. New-primary receives DVC quorum. Suffix is 6,7,8.
                        //    It already has 6,7. op=8 is new to us, but part of our prior log_view.
                        // 3. New-primary repairs 8 (writes it to the WAL).
                        // 4. New-primary crashes/recovers — op=8 was part of the same log_view, so the
                        //    journal doesn't truncate it, and the primary recovers with a new head.
                        // To avoid special-casing this all over, we pretend this higher op doesn't
                        // exist. This is safe because the prior view-change didn't complete.
                    }
```

might signal clearer that comment belongs to the empty line, but it's not really.